src/stepfunctions/steps/sagemaker.py

```diff
@@ -103,6 +116,16 @@ def __init__(self, state_id, estimator, job_name, data=None, hyperparameters=Non
         if isinstance(job_name, (ExecutionInput, StepInput)):
             parameters['TrainingJobName'] = job_name
 
+        if output_path is not None:
+            if isinstance(output_path, (ExecutionInput, StepInput)) or isinstance(output_path, str):
```

Comment:
This silently ignores invalid types. We should either remove this type check and let things fail downstream or throw an error if output_path is not a str or Placeholder

Reply:
Right, we definitely don't want to ignore those! Will remove the checks.